### PR TITLE
Wppf fix

### PR DESCRIPTION
**Pull request by saransh13**
1. Lots of duplicate code removed from the WPPF code. A lot more to remove, but will be done as part of a future PR. Don't want to stray too far from master
2. LeBail now fully supports masked arrays. Changes propagated to the `extract_intensities' function too. Masked arrays are a fair bit slower than just the numpy arrays, but that is a problem to be tackled another day.
3. The `Material` class needed a few new properties to support changes in the `WPPF` module.

### CODE REVIEW

File: hexrd/WPPF.py [hunk 1]
```diff
@@ -4815,6 +4642,139 @@ def Refine(self):
...
+                        nz = f"{p}_{elem}{atom_label[i]}_z"
+                        oc = f"{p}_{elem}{atom_label[i]}_occ"
+
+                        if mat.aniU:
+                            Un = []
+                            for j in range(6):
+                                Un.append(
+                                    f"{p}_{elem}"
+                                    "{atom_label[i]}"
+                                    "_{_nameU[j]}")
```
Comment (psavery) [inline hexrd/WPPF.py:4719]: These lines need to start with `f`

File: hexrd/WPPF.py [hunk 2]
```diff
@@ -4899,6 +4859,148 @@ def scale(self, value):
...
+        """
+        dictionary of materials class
+        """
+        for k, m in mat.items():
+            _add_atominfo_to_params(params, m)
+
+    else:
+        msg = f"_generate_default_parameters: "
+        "incorrect argument. only list, dict or "
+        "Material is accpeted."
```
Comment (psavery) [inline hexrd/WPPF.py:4917]: Need to add parentheses here

File: hexrd/WPPF.py [hunk 3]
```diff
@@ -4899,6 +4859,148 @@ def scale(self, value):
...
+        nn = f"{phase_name}_{elem}{atom_label[i]}_occ"
+        params.add(nn, value=occ[i],
+                   lb=0.0, ub=1.0,
+                   vary=False)
+
+        if(mat.aniU):
+            U = mat.U
+            for j in range(6):
+                nn = f"{phase_name}_{elem}{atom_label[i]}"
+                "_{nameU[j]}"
```
Comment (psavery) [inline hexrd/WPPF.py:4990]: Add parentheses and `f` at the beginning of the second line.

**Review by psavery — COMMENTED** [review]
Removing duplicated code is a big positive. Looking nice!

File: hexrd/WPPF.py (hunk 3 above)
Comment (saransh13) [reply hexrd/WPPF.py]: Fixed.

File: hexrd/WPPF.py (hunk 2 above)
Comment (saransh13) [reply hexrd/WPPF.py]: Done.

File: hexrd/WPPF.py [hunk 4]
```diff
@@ -4345,11 +4263,29 @@ def initialize_bkg(self):
...
+            #     yy = np.squeeze(snip1d_quad(np.atleast_2d(s.y),
+            #                                 w=ww, numiter=numiter))
+            #     self._background.append(Spectrum(x=self._tth_list[i], y=yy))
+
+            def chebyshevfit(self):
+                degree = self.bkgmethod['chebyshev']
+                p = np.polynomial.Chebyshev.fit(
+                    self.tth_list, self.spectrum_expt._y,
+                    degree, w=self.weights**2)
+                self.background = Spectrum(x=self.tth_list, y=p(self.tth_list))
```
[the remark this answers is not in the document]
Comment (saransh13) [reply hexrd/WPPF.py]: Yep, definitely indented too far.

File: hexrd/WPPF.py [hunk 5]
```diff
@@ -1747,8 +1752,8 @@ def selectpoints(self):
         """
         self.points = []
         for i, s in enumerate(self._spectrum_expt):
-            txt = f'Select points for background estimation; \
-                click middle mouse button when done. segment # {i}'
+            txt = (f"Select points for background estimation;"
+                "click middle mouse button when done. segment # {i}")
```
[the remark this answers is not in the document]
Comment (saransh13) [reply hexrd/WPPF.py]: Fixed it.

**Review by joelvbernier — APPROVED** [review]

**Pull request merged**